### PR TITLE
chore: make cost factor configurable

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Introduce some settings to configure the cost factors.

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14445)

### CODE REVIEW

**Review by andylokandy — APPROVED** [review]

File: src/query/settings/src/settings_default.rs [hunk 1]
```diff
@@ -624,6 +631,24 @@ impl DefaultSettings {
                     mode: SettingMode::Both,
                     range: Some(SettingRange::Numeric(0..=1)),
                 }),
+                ("cost_factor_hash_table_per_row", DefaultSettingValue {
+                    value: UserSettingValue::UInt64(COST_FACTOR_HASH_TABLE_PER_ROW),
+                    desc: "Cost factor of building hash table for a data row",
+                    mode: SettingMode::Both,
+                    range: Some(SettingRange::Numeric(0..=u64::MAX)),
+                }),
+                ("cost_factor_aggregate_per_row", DefaultSettingValue {
```
Comment (BohuTANG) [inline src/query/settings/src/settings_default.rs:640]: These new settings is for developer or the user? The `desc` is not easy to understand how to use them.

Comment (leiysky) [reply src/query/settings/src/settings_default.rs:640]: Yes, it’s for developers. Do we have solution not to expose these settings to users?

Comment (BohuTANG) [reply src/query/settings/src/settings_default.rs:640]: Yes, we don't recommand export the `VALUE` setting, if they set the global value it hard to adjust it in the future. We should atomic setting them by us not setting.

Comment (leiysky) [reply src/query/settings/src/settings_default.rs:640]: > Settings is good for: disable or enable a feature by us.

I need a way to modify the configuration dynamicly to tune the cost model. Do you have any suggestion for this?

Comment (BohuTANG) [reply src/query/settings/src/settings_default.rs:640]: Ok, I see. This PR works.

Comment (BohuTANG) [reply src/query/settings/src/settings_default.rs:640]: BTW, we can not tell the user to set these settings, only used by us.

**Pull request merged**